rstudio/rstudio
# Improve Visual Editor Cross Ref Completions

### Intent

Improve the visual editor support for automatic completion. In particular, improve mark recognition when adjacent to punctuation and add support for Quarto docs.

### Approach

Add new plugins to support additional completions. 

### Automated Tests

N/A

### QA Notes

Should not change existing behavior- should only be additive. Cites that are directly adjacent to a . or ? (e.g. `See @knuth1984.`) will be properly handled (previously the punctuation would be considered a part of the citation).

### Checklist
